settings/src/components/account-status.js

```diff
@@ -21,12 +21,20 @@ export default function AccountStatus() {
 	const totpStatus = record[ '2fa_available_providers' ].includes( 'Two_Factor_Totp' )
 		? 'enabled'
 		: 'disabled';
+	const primaryProviderEnabled = 'enabled' === totpStatus; // @todo Update when webauthn UI built.
```

Comment:
Since `totpStatus` is passed into `<SettingStatusCard>` as a string but used in other cases as a boolean, doesn't it make sense to change that logic? 

```
const primaryProviderEnabled = record[ '2fa_available_providers' ].includes( 'Two_Factor_Totp' );

....

<SettingStatusCard
  screen="totp"
  status={ primaryProviderEnabled ? 'enabled' : 'disabled' }
  headerText="Two-Factor Authentication"
  bodyText={
	  primaryProviderEnabled
		  ? /* @todo update this when hardware tokens become an additional option. */
		    'You have two-factor authentication enabled using an app.'
		  : 'You do not have two-factor authentication enabled.'
  }
/>

```

Reply:
🤔 I don't think so, the `totp` status card needs to use `totpStatus` specifically, because if WebAuthn is enabled and TOTP is disabled, then `primaryProviderEnabled` would be `true` and `totpStatus` would be`'disabled'`. It looks like the code above would show the wrong status in that situation.

I added 747921c82a9eba07bdfb346cf9b6dc1273cd4fd8 to clarify how I imagine this will be used once we add WebAuthn.

Does that make sense, or did I misunderstand something?

Reply:
Right, I understand how my code sample doesn't work. The main point is that it may be better to assign the status as a boolean since we are making checks based on it's string as opposed to a more predictable boolean.

```
const totpEnabled = record[ '2fa_available_providers' ].includes( 'Two_Factor_Totp' );
const webAuthnEnabled= record[ '2fa_available_providers' ].includes( 'TwoFactor_Provider_WebAuthn');
const primaryProviderEnabled = totpEnabled || webAuthnEnabled;

...

<SettingStatusCard
  screen="totp"
  status={ totpEnabled ? 'enabled' : 'disabled' }
  headerText="Two-Factor Authentication"
  bodyText={
	  totpEnabled
		  ? /* @todo update this when hardware tokens become an additional option. */
		    'You have two-factor authentication enabled using an app.'
		  : 'You do not have two-factor authentication enabled.'
  }
/>


```

Reply:
That WFM, changed in 4ccb203 👍🏻